src/components/AnimateSharedLayout/utils/stack.ts

```diff
@@ -101,7 +101,7 @@ export function layoutStack(): LayoutStack {
             })
 
             if (
-                state.follow &&
+                state.lead !== prevState.follow &&
```

Comment:
Can you explain this change? I'm having a hard time understanding what it solves.

Reply:
Yeah this is basically when we're animating the lead back to a follow, then when thats finished AnimatePresence rerenders, removes the lead, and makes the old follow the new lead. I'll add a comment tho